File: market-mapper/src/main/java/by/market/mapper/DoubleCharacteristicMapper.java
package by.market.mapper;

import by.market.domain.characteristics.single.DoubleCharacteristic;
import by.market.dto.characteristics.DoubleCharacteristicDTO;
import org.mapstruct.InheritInverseConfiguration;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(config = MapperConfiguration.class)
public interface DoubleCharacteristicMapper extends MapstructMapper<DoubleCharacteristicDTO, DoubleCharacteristic> {

    @Override
    @Mapping(source = "characteristic.id", target = "characteristic")
    @Mapping(source = "product.id", target = "product")
    DoubleCharacteristicDTO toMap(DoubleCharacteristic objectAbstractCharacteristic);

    @Override
    @InheritInverseConfiguration
    DoubleCharacteristic fromMap(DoubleCharacteristicDTO objectAbstractCharacteristicDTO);

}

File: market-mapper/src/main/kotlin/by/market/mapper/characteristics/single/DoubleCharacteristicMapper.kt
package by.market.mapper.characteristics.single

import by.market.domain.characteristics.single.DoubleCharacteristic
import by.market.dto.characteristics.DoubleCharacteristicDTO
import by.market.mapper.IMapstructMapper
import by.market.mapper.MapperConfig
import org.mapstruct.Mapper

@Mapper(config = MapperConfig::class)
interface DoubleCharacteristicMapper : IMapstructMapper<DoubleCharacteristicDTO, DoubleCharacteristic>
